src/dvsim/utils/wildcards.py

```diff
@@ -0,0 +1,375 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Wildcard substitution in strings within data structures.
+
+DVSim configuration relies heavily on wildcard based substitutions and dynamic
```

Comment:
I think that "wildcard" is a bit of a term of art from Sri. Maybe this could be written as
```
DVSim configuration relies heavily on dynamic templating and substitutions based on "wildcards".
```
?